src/main/java/com/github/onsdigital/babbage/search/helpers/SearchHelper.java

```diff
@@ -67,8 +74,31 @@ public static ONSSearchResponse search(ONSQuery queries) {
                 if (item.isFailure()) {
                     throw new ElasticsearchException(item.getFailureMessage());
                 }
-                helpers.add(resolveDetails(queries.get(i), new ONSSearchResponse(item.getResponse())));
-                i++;
+                ONSQuery query = queries.get(i);
+                ONSSearchResponse searchResponse = resolveDetails(query, new ONSSearchResponse(item.getResponse()));
+
+                List<ContentType> types = Arrays.asList(query.types());
+
+                // Check for retired product pages
+                if (query.size() == 1 &&
```

Comment:
is it possible for either types or query to be null? If so you need to add a defensive check

Reply:
Types could be null but query should never be null here (if it is the query to ES will fail in the first place)